pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -0,0 +1,191 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from time import sleep
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    strict_range,
+    truncated_range,
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class CNT91(Instrument):
+    """Represents a Pendulum CNT-91 frequency counter."""
+
+    CHANNELS = {"A": 1, "B": 2, "C": 3, "E": 4, "INTREF": 6}
+    MAX_BUFFER_SIZE = 32000  # User Manual 8-38
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(resourceName, "Pendulum CNT-91", **kwargs)
+        self.adapter.connection.timeout = 120000
+
+    def ask(self, command):
+        """
+        Writes the command to the instrument through the adapter
+        and returns the read response.
+
+        :param command: command string to be sent to the instrument
+        """
+        # Infinite loop occures if not redefined.
+        return super().ask(command).rstrip("\n")
```

Comment:
instead of using the `.rstrip()` here, it might be better to set `read_termination = "\r\n",` as part of the `super().__init()` call.
 more details on the terminations [here](https://pyvisa.readthedocs.io/en/latest/introduction/communication.html#getting-the-instrument-configuration-right)

Reply:
Thanks! Missed that part of the documentation. This is indeed much cleaner. Fixed in 5e2e222.